packages/material/src/complex/table-array.control.tsx

```diff
@@ -41,117 +55,325 @@ export const tableArrayTester: RankedTester = rankWith(10, and(
+    const { uischema, schema, path, data, visible, childErrors } = this.props;
     const controlElement = uischema as ControlElement;
-
     const tableClass = JsonForms.stylingRegistry.getAsClassName('array-table.table');
-    const labelClass = JsonForms.stylingRegistry.getAsClassName('array-table.label');
-    const buttonClass = JsonForms.stylingRegistry.getAsClassName('array-table.button');
     const controlClass = [JsonForms.stylingRegistry.getAsClassName('array-table'),
-      Helpers.convertToValidClassName(controlElement.scope.$ref)].join(' ');
+    Helpers.convertToValidClassName(controlElement.scope.$ref)].join(' ');
 
     const resolvedSchema = resolveSchema(schema, controlElement.scope.$ref + '/items');
-    const createControlElement = (key: string): ControlElement => ({
-      type: 'Control',
-      label: false,
-      scope: { $ref: `#/properties/${key}` }
-    });
-    const labelObject = Helpers.createLabelDescriptionFrom(controlElement);
-    const isValid = errors.length === 0;
-    const divClassNames = 'validation' + (isValid ? '' : ' validation_error');
+
+    const handleSelectAllClick = (_event, checked) => {
+      if (checked) {
+        this.setState({ selected: data });
+
+        return;
+      }
+      this.setState({ selected: [] });
+    };
+
+    const handleClick = (_event, child) => {
+      const selected = this.state.selected.filter(s => data.indexOf(s) !== -1);
+      const selectedIndex = selected.indexOf(child);
+      let newSelected = [];
+
+      if (selectedIndex === -1) {
+        newSelected = newSelected.concat(selected, child);
+      } else if (selectedIndex === 0) {
+        newSelected = newSelected.concat(selected.slice(1));
+      } else if (selectedIndex === selected.length - 1) {
+        newSelected = newSelected.concat(selected.slice(0, -1));
+      } else if (selectedIndex > 0) {
+        newSelected = newSelected.concat(
+          selected.slice(0, selectedIndex),
+          selected.slice(selectedIndex + 1),
+        );
+      }
+
+      this.setState({ selected: newSelected });
+    };
+
+    const isSelected = child => this.state.selected.indexOf(child) !== -1;
+    const handleOpenConfirmDelete = () => {
+      this.setState({ openConfirmDelete: true });
+    };
+
+    const handleClose = () => {
+      this.setState({ openConfirmDelete: false });
+    };
+
+    const isEmptyTable = !data || !Array.isArray(data) || data.length === 0;
+    const tableProps = {data, path, resolvedSchema, childErrors,
+        isSelected, handleClick, handleOpenConfirmDelete};
+    const numSelected = this.state.selected.length;
+    const rowCount = data.length;
+    const toolbarProps = {
+      errors: this.props.errors,
+      childErrors,
+      label: this.props.label,
+      path,
+      uischema,
+      numSelected,
+      handleOpenConfirmDelete,
+      dispatch: this.props.dispatch
+    };
+    const removeItems = () => {
+      this.props.dispatch(
```

Comment:
Extract to `mapDispatchToProps` (analogously to default export of `ArrayControlRenderer.tsx`). I think it would be beneficial to follow the same structure as with the vanilla renderer, meaning that we do the split between the presentational component and the container component.